rp2040-hal/src/spi.rs

```diff
@@ -251,15 +252,32 @@ impl<S: State, D: SpiDevice, P: ValidSpiPinout<D>, const DS: u8> Spi<S, D, P, DS
     }
 }
 
-impl<D: SpiDevice, P: ValidSpiPinout<D>, const DS: u8> Spi<Disabled, D, P, DS> {
-    /// Create new not initialized Spi bus. Initialize it with [`.init`][Self::init]
+impl<D: SpiDevice, P: ValidSpiPinout<D>> Spi<Disabled, D, P, 8> {
+    /// Create new (not initialized) Spi bus with the default data size (8 bits).
+    ///
+    /// Initialize it with [`.init`][Self::init]
     /// or [`.init_slave`][Self::init_slave].
     ///
     /// Valid pin sets are in the form of `(Tx, Sck)` or `(Tx, Rx, Sck)`
     ///
     /// If you pins are dynamically identified (`Pin<DynPinId, _, _>`) they will first need to pass
     /// validation using their corresponding [`ValidatedPinXX`](ValidatedPinTx).
-    pub fn new(device: D, pins: P) -> Spi<Disabled, D, P, DS> {
+    pub fn new(device: D, pins: P) -> Spi<Disabled, D, P, 8> {
```

Comment:
It seems to me that choosing 8bits for the `fn new` a bit arbitrary.
I understand 8 bits were used as the default for the type before, but I don't think it makes it more correct 😛

Reply:
To be fair I don't have that much embedded experience, but it seems like 8 bits as word size is extremely common in general and would at least not be particularly surprising

Reply:
That was my assumption as well, but to be honest my experience with embedded peripherals is also quite limited.

Reply:
I had a look at embassy-rp to see how they handle the word width. They don't. It's hard coded to 8 bits. :smile: 
https://github.com/embassy-rs/embassy/blob/main/embassy-rp/src/spi.rs#L87